go-controller/pkg/ovn/ovn.go

```diff
@@ -162,46 +65,13 @@ type Controller struct {
 	// libovsdb southbound client interface
 	sbClient libovsdbclient.Client
 
-	// retry framework for pods
-	retryPods *retry.RetryFramework
-
-	// retry framework for network policies
-	retryNetworkPolicies *retry.RetryFramework
-
-	// retry framework for egress firewall
-	retryEgressFirewalls *retry.RetryFramework
-
-	// retry framework for egress IP
-	retryEgressIPs *retry.RetryFramework
-	// retry framework for egress IP Namespaces
-	retryEgressIPNamespaces *retry.RetryFramework
-	// retry framework for egress IP Pods
-	retryEgressIPPods *retry.RetryFramework
-	// retry framework for Egress nodes
-	retryEgressNodes *retry.RetryFramework
-	// EgressIP Node-specific syncMap used by egressip node event handler
-	addEgressNodeFailed sync.Map
-
-	// retry framework for nodes
-	retryNodes *retry.RetryFramework
-	// Node-specific syncMaps used by node event handler
-	gatewaysFailed              sync.Map
-	mgmtPortFailed              sync.Map
-	addNodeFailed               sync.Map
-	nodeClusterRouterPortFailed sync.Map
-	hybridOverlayFailed         sync.Map
-
-	// retry framework for Cloud private IP config
-	retryCloudPrivateIPConfig *retry.RetryFramework
-
-	// retry framework for namespaces
-	retryNamespaces *retry.RetryFramework
-
-	podRecorder metrics.PodRecorder
+	// has SCTP support
+	SCTPSupport bool
+}
 
-	// variable to determine if all pods present on the node during startup have been processed
-	// updated atomically
-	allInitialPodsProcessed uint32
+type NetworkController interface {
+	Start(ctx context.Context) error
+	Stop()
```

Comment:
Since this interface is used and imposed by NetworkControllerManager, I would move it to that package and file: `network_controller_manager.go`. the ovn package has no use of this interface, since Start or Stop are never called form the ovn package.
As I suggested to remove `oc` from masterEventHandler as well, we don't have a use for it there either.